src/blas/backends/rocblas/rocblas_level3.cpp

```diff
@@ -381,10 +381,19 @@ inline void trmm(Func func, sycl::queue &queue, side left_right, uplo upper_lowe
             auto a_ = sc.get_mem<rocDataType *>(a_acc);
             auto b_ = sc.get_mem<rocDataType *>(b_acc);
             rocblas_status err;
+
+// rocblas version 4.0.0 removed the legacy BLAS trmm implementation
+#ifdef ROCBLAS_NO_LEGACY_TRMM
```

Comment:
```suggestion
#if ROCBLAS_VERSION_MAJOR >= 3
```
We can use `ROCBLAS_VERSION_MAJOR`, that way we don't need to add a new define, also since it's deprecated in rocBLAS 3 it gives warning so we could also use the new interface for rocBLAS 3.x.

I ran into the same issue and ended up with a similar patch before finding this PR. I didn't have any `ROCBLAS_LIB_PATH` issues with rocBLAS 4.x though.